File: 001.Two-Sum/python/001.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        #hash解法
        
        d = {}
        
        for i in range(len(nums)):
            find = target - nums[i]
            if d.get(find,None) == None:
                d[nums[i]] = i  #创建字典，数为key，值为下标
            else:
                return [d[find],i]
            
        

File: 001.Two-Sum/rust/001.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::with_capacity(nums.len());
        for(index,num) in nums.iter().enumerate() {
            match map.get(&(target-num)) {
                None => {map.insert(num,index);}
                Some(find) => {return vec![*find as i32, index as i32]}
            }
        } 
        vec![]
    }
}